components/google-cloud/google_cloud_pipeline_components/experimental/remote/gcp_launcher/batch_prediction_job_remote_runner.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2021 The Kubeflow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+from os import path
+import time
+
+from google.api_core import gapic_v1
+from google.cloud import aiplatform
+from google.cloud.aiplatform.compat.types import job_state as gca_job_state
+from google.protobuf import json_format
+from google_cloud_pipeline_components.experimental.proto.gcp_resources_pb2 import GcpResources
+
+_POLLING_INTERVAL_IN_SECONDS = 20
```

Comment:
It would be really nice if you could refactor out the common logics and share with custom job. 
The same code pattern can be used for hp tuning job in the near future. 

optionally you can leave a todo in this file to me.

Reply:
Done, please let me know how it looks.